bin/zis-parmlib-update.sh

```diff
@@ -0,0 +1,284 @@
+#!/bin/sh
+# This program and the accompanying materials are
+# made available under the terms of the Eclipse Public License v2.0 which accompanies
+# this distribution, and is available at https://www.eclipse.org/legal/epl-v20.html
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+# Copyright Contributors to the Zowe Project.
+
+# This script adds or updates pramaters in ZIS PARMLIB member specified in
+# the provided SAMPLIB file.
+
+# Make sure all files created in this script are only accesible by the owner.
+umask go-rwx
+
+# Prohibit undefined variables.
+set -u
+
+# ZIS-related global variables.
+ZIS_PREFIX=${ZIS_PREFIX:-$USER.DEV}
+ZIS_PARMLIB=${ZIS_PARMLIB:-$ZIS_PREFIX.PARMLIB}
+ZIS_PARMLIB_MEMBER=${ZIS_PARMLIB_MEMBER:-ZWESIP00}
+
+#################################################
+# Generate a name for a temporary file in /tmp
+#
+# Globals:
+#   RANDOM
+# Arguments:
+#   None
+# Returns:
+#   The generated file name
+#################################################
+make_temp_filename() {
+  name=/tmp/zis-sample-install-$$-$RANDOM
+  while [ -e $name ]
+  do
+    name=/tmp/zis-sample-install-$$-$RANDOM
+  done
+  echo $name
+}
+
+#################################################
+# Try to resolve values that are defined using
+# environmental variables, otherwise return
+# the original value
+#
+# Globals:
+#   None
+# Arguments:
+#   $1 Value from a key-value pair
+# Returns:
+#   * If an env variable is provided, its value
+#     is returned on success
+#   * If an env varible if provided and
+#     the variable is not defined,
+#     string VALUE_NOT_FOUND is returned
+#   * The original value is returned
+#################################################
+resolve_env_parameter() {
+
+  parm=$1
+
+  # Are we dealing with an env variable based value?
+  echo $parm | grep -E "^[$]{1}[A-Z0-9_]+$" 1>/dev/null
+  # Yes, resolve the value using eval, otherwise return the value itself.
+  if [ $? -eq 0 ]; then
```

Comment:
Simplification:
```
if echo $parm | grep -E -q "^[$]{1}[A-Z0-9_]+$"; then
  ...
```

Reply:
My check seems more readable to me.

Reply:
I think Fyodor may have a good point here :)

Reply:
2:1, you win.

Reply:
IIRC, ```if``` is a command so always returns a 0 return code so it's a bug

Reply:
No, this check should be correct. ```if``` executes the command first and checks the return code, it should not affect ```$?``` of ```grep```, I double-checked.

Reply:
I haven't set up any ERR traps.

Reply:
My bad. We save the return code because we print it in the message and it gets clobbered by echo. Sorry!

Reply:
Ah, ok. Though, I'm still not sure why we need two nested subshells: the parenthesis are one, and the `eval` is another. `(set -u && echo $foo) 2>/dev/null` seems to be enough

Reply:
Add an error trap and used an inline if check.